### PR TITLE
199 active learning docstring examples

### LINKED ISSUE #199: Add examples to docstrings
Some of the modules could be better explained. Sometimes it helps a lot to see sample code. I would propose to add sample code for the following modules in the documentation:

- [x] `lightly.api.download.get_samples_by_tag`
- [x] `lightly.active_learning.agents.agent.ActiveLearningAgent` + `query` method
- [x] `lightly.active_learning.config.sampler_config.SamplerConfig`

### PR DESCRIPTION
**Pull request by philippmwirth**
# 199 active learning docstring examples

Closes #199.

- Deprecated the old way to download filenames
- Added examples to the docstrings of `ActiveLearningAgent` and `SamplerConfig`
- Fixed the imports, the following are now possible
```python
from lightly.active_learning.agents import ActiveLearningAgent
from lightly.active_learning.scorers import ScorerClassification
from lightly.active_learning.scorers import ScorerObjectDetection
from lightly.active_learning.config import SamplerConfig
```

### CODE REVIEW

File: lightly/active_learning/config/sampler_config.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ class SamplerConfig:
 
     Attributes:
         method:
-            The method to use for sampling, e.g. CORESET.
+            The method to use for sampling, one of CORESET, RANDOM, CORAL
```
Comment (MalteEbner) [inline lightly/active_learning/config/sampler_config.py:11]: What about ACTIVE_LEARNING?

**Review by MalteEbner — APPROVED** [review]

File: lightly/active_learning/config/sampler_config.py (hunk 1 above)
Comment (philippmwirth) [reply lightly/active_learning/config/sampler_config.py:11]: Good point

File: lightly/active_learning/config/sampler_config.py [hunk 2]
```diff
@@ -21,6 +21,17 @@ class SamplerConfig:
             other attributes and the datetime. A new tag will be created in the
             web-app under this name.
 
+    Examples:
+        >>> # sample 100 images with CORESET sampling
+        >>> config = SamplerConfig(method=SamplingMethod.CORESET, n_samples=100)
+        >>> config = SamplerConfig(method='CORESET', n_samples=100)
```
Comment (philippmwirth) [inline lightly/active_learning/config/sampler_config.py:27]: @MalteEbner do you know if this actually works?

Comment (MalteEbner) [reply lightly/active_learning/config/sampler_config.py:27]: Yes, it looks well.

**Pull request merged**